CMakeLists.txt

```diff
@@ -577,6 +583,17 @@ if (ENABLE_SHARED)
 	foreach (tar srt_virtual haicrypt_virtual)
 		set_target_properties(${tar} PROPERTIES POSITION_INDEPENDENT_CODE 1)
 	endforeach()
+
+	#add resource files to shared library, to set DLL metadata on Windows DLLs
+	if (WIN32 AND NOT MINGW)
```

Comment:
That is true... but I didn't actually write that line :)

Are you asking me to try and change these other elements as part of my PR because you'd just like it cleaned up?

I'm happy to give it a whirl... but I just wanted to be clear this does not come from me.

Reply:
I just checked and that shortcut also appears with a total of 6 times in that list... so actually, if you want that changed, I'd prefer that was done under a different PR rather than bog this one down.

Reply:
I don't understand. This line I marked here is marked as a change introduced by this PR. Should that be not true, please double check or possibly update the baseline. Here's the fragment around there at the current master:
```
if (ENABLE_SHARED)
	# Set this to sources as well, as it won't be automatically handled
	foreach (tar srt_virtual haicrypt_virtual)
		set_target_properties(${tar} PROPERTIES POSITION_INDEPENDENT_CODE 1)
	endforeach()
endif()

# Manual handling of dependency on virtual library
# By setting the target, all settings applied to the haicrypt target
```

Shall we have any misunderstanding: I mean to change `if (WIN32 AND NOT MINGW)` into `if (MICROSOFT)`.